tests/test_relations.py

```diff
@@ -342,6 +342,49 @@ def get_queryset(self):
         field.to_internal_value(self.instance.name)
 
 
+class TestNestedSlugRelatedField(APISimpleTestCase):
+    def setUp(self):
+        self.queryset = MockQueryset([
+            MockObject(pk=1, name='foo', nested=MockObject(pk=2, name='bar')),
+            MockObject(pk=3, name='hello', nested=MockObject(pk=4, name='world')),
+            MockObject(pk=5, name='harry', nested=MockObject(pk=6, name='potter'))
+        ])
+        self.instance = self.queryset.items[2]
+        self.field = serializers.SlugRelatedField(
+            slug_field='nested__name', queryset=self.queryset
```

Comment:
can you add some more relevant test case in this regard?

Reply:
I have added more test cases regarding that.